server/src/org/apache/cloudstack/network/topology/BasicNetworkVisitor.java

```diff
@@ -316,4 +317,9 @@ public boolean visit(final StaticRoutesRules staticRoutesRules) throws ResourceU
     public boolean visit(final AdvancedVpnRules vpnRules) throws ResourceUnavailableException {
         throw new CloudRuntimeException("AdvancedVpnRules not implemented in Basic Network Topology.");
     }
+
+    @Override
+    public boolean visit(final QuaggaRules vpnRules) throws ResourceUnavailableException {
+        throw new CloudRuntimeException("AdvancedVpnRules not implemented in Basic Network Topology.");
```

Comment:
Consider using the ``OperationNotSupported`` exception provided by the Java runtime for this purpose.